### DISCUSSION TITLE
Hide env defined apikeys

### DISCUSSION THREAD

**Discussion by sebiweise** [open-webui/open-webui#2020] [General]
# Bug Report

## Description

**Bug Summary:**
When I define api keys via env vars, the vars can be seen by the users in there settings.

**Steps to Reproduce:**
Define api keys in env vars

**Expected Behavior:**
The api keys should not be shown once they are saved.

**Actual Behavior:**
The api keys can be shown once saved or loaded via env vars.

## Environment

- **Open WebUI Version:** v0.1.123

- **Operating System:** Ubuntu 22.04
- **Browser (if applicable):** Chrome

## Reproduction Details

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.

## Logs and Screenshots

**Screenshots (if applicable):**
![image](https://github.com/open-webui/open-webui/assets/22999200/13cf3b9c-a2b7-49cd-bb9e-0593f0de094e)

## Installation Method

I´ve installed open-webui via Git and the Dockerfile on my Coolify instance

**Comment by tjbck** [comment]
They're only visible for users with admin role for editing purposes. I personally don't see this being an issue but I'd love to hear what other people think.

**Comment by JuanjoJG-WATA** [comment]
I agree with @tjbck. Only admins can see it, and if they are admins in your system it means you trust them enough to know or handle with the keys.

**Comment by sebiweise** [comment]
Oh okey, I haven´t checked with a user, I just checked with another Admin account. Yes then your right, as long as the users can´t see the keys its totally fine, sorry for that.

**Comment by johannoriel** [comment]
I think it should be hidden. It's standard security practice : every password should not be display by default, because... there can be someone looking at the screen. We cannot assume OpenWebui is only used by admin alone in their room.
The "editing" purpose is very rare because most of the time we only paste the key from elsewhere. And if someone really need to edit a key, there are plugin for that, but it's good practice not to be easy to do.

**Comment by shishiraiyar** [comment]
My two cents @JuanjoJG-WATA: It would be nice if admins had "write-only" access to the keys. i.e. admins can go to the settings and set a new key, but once it's set, they can't see it. This is how secrets are handled by GitHub actions for example.
